src/core/Entry.h

```diff
@@ -232,6 +235,8 @@ private slots:
     EntryAttachments* const m_attachments;
     AutoTypeAssociations* const m_autoTypeAssociations;
 
+    CustomData* const m_customData;
```

Comment:
I'm not sure, why `QPointer` should be used since these objects are managed by the qt-parent relationship. Do you have some general guideline when to use the one memory management mechanism over the other?

Reply:
Don't confuse QPointer with QSharedPointer or QScopedPointer. QPointer is simply a guard that resets the pointer to nullptr whenever the object behind it is deleted, so we never get dangling pointers. It's a precaution to prevent bugs.